includes/connect/class-wc-stripe-connect.php

```
@@ -549,6 +554,47 @@ public function refresh_connection() {
+		 * @param WP_Error $response The terminal error returned by the refresh request.
+		 */
+		private function handle_terminal_refresh_failure( string $prefix, WP_Error $response ): void {
+			// The refresh token will never come back, so do not re-arm the periodic refresh.
+			$this->unschedule_connection_refresh();
+
+			// Mark the attempt budget as exhausted to keep parity with the "gave up after 10
+			// tries" state the rest of the account-status messaging already expects.
+			update_option( 'wc_stripe_' . $prefix . 'oauth_failed_attempts', 10 );
+			update_option( 'wc_stripe_' . $prefix . 'oauth_last_failed_at', time() );
+
+			WC_Stripe_Logger::error( 'OAuth connection refresh failed terminally; reconnection required.', [ 'response' => $response ] );
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'handle_terminal_refresh_failure'. (undefined)

(StaticAccess)